components/sensor/ppd42x.rst

```diff
@@ -0,0 +1,80 @@
+PPD42X Particulate Matter Sensor
+================================
+
+.. seo::
+    :description: Instructions for setting up PPD42X Particulate matter sensors
+    :image: ppd42x.jpg
+    :keywords: PPD42X
+
+The ``PPD42X`` sensor platform allows you to use PPD42X particulate matter sensors (`datasheet <https://www.researchgate.net/profile/Visa_Tasic/project/low-cost-sensor-platform-for-air-quality-monitoring/attachment/58332d5908ae91d0fe260283/AS:430912497295360@1479748953328/download/ShinyeiPPD42NS_TechDoc.pdf?context=ProjectUpdatesLog>`__)
+with ESPHome.
+
+.. figure:: images/ppd42-full.jpg
+    :align: center
+    :width: 50.0%
+
+    PPD42X Particulate Matter Sensor
+Wiring:
+Pin Function    Wire color (in picture above)
+1   Ground	                   Green
+2   P2 (output)                White
+3   5V                         Yellow
+4   P1 (output)                Black
+5   Threshold voltage for P2   Red
+
+Pin 5 is closest to the corner of the board (and there is a 5 printed on both sides of the board near pin 5).
+Note the colors in the photo above are different from the wire colors available from other sources (like Grove).
+
+The P1 output is used to measure particles between 1um and 10um.
+The P2 output (with the threshold pin unconnected) is used to measure particles between 2.5um and 10um.
+To read the sensor, the total time that the P1 or P2 signal is low (measured in microsends) is used to determine particle concentrations.
+
+.. code-block:: yaml
+
+    # Example configuration entry
+
+    sensor:
+      - platform: ppd42x
+        time_out: 30000
+        pm_2_5:
+          name: "Particulate Matter <2.5µm Concentration"
+        pm_10_0:
+          name: "Particulate Matter <10.0µm Concentration"
+        update_interval: 5min
+
+With ``update_interval``, the working period of the PPD42X device will be changed. If ``update_interval`` is
+equal to ``0min``, the PPD42X will be set to continuous measurement and will report new measurement values
+approximately every second.
+
+If ``update_interval`` is between 1-30 minutes, the PPD42X periodically turns on for 30s before each measurement.
+For the remaining time the sensor is shut off. As a result, this mode can reduce power consumption and increases
+the lifetime of the PPD42X.
+
+Configuration variables:
+------------------------
+- **time_out** (*mandatory*, :ref:`config-time`): The interval to check the P1 and P2 pins in micro-seconds.
```

Comment:
required not mandatory - and it *should* not be mandatory, it should be an optional variable and moved to the bottom.

Reply:
ok , will change it in the code too